std/concurrency.d

```diff
@@ -693,18 +693,18 @@ body
 
 unittest
 {
-    assert( __traits( compiles,
+    static assert( __traits( compiles,
```

Comment:
Maybe we should probably change `__traits(compiles, ...)` to a proper unittest. There was a call to action about that not long ago on the newsgroup. Anyway this change is an improvement.